tidy3d/components/data/monitor_data.py

```
@@ -811,6 +817,19 @@ def _interpolated_tangential_fields(self, coords: ArrayFloat2D) -> dict[str, Dat
         """
         fields = self._tangential_fields
 
+        try:
+            # If coords already match, just return the tangential fields directly.
+            # Using try: except for flow control.
+            for field in fields.values():
+                for idim, dim in enumerate(self._tangential_dims):
+                    if field.coords[dim].values.size != coords[idim].size or not np.all(
+                        field.coords[dim].values == coords[idim]
+                    ):
+                        raise ValueError("Coordinates do not match.")
+            return fields
+        except ValueError:
+            pass
```

Also here I think bool logic is preferable. From my understand raising an exception can also be slower. In this case, the interpolation is the expensive part so not a big problem, but good to avoid I think.

---

A method that can determine whether two monitor data have matching tangential fields would be useful when I get back to the `dot_yee`/`flux_yee` stuff as well

---

I could switch this to bool and split this out in a separate method. What do you think about the other one? There I'm more worried about something erroring that I haven't anticipated (like e.g. due to dimensions order) so I kind of want it to fallback to the xarray handling.

---

Well I guess that would indicate a bug we need to fix. I am trying to think how it could fail, from your check the `coords` must be identical right? Same shape and order I would think.

---

Oh I see, only the tangential dims are checked

---

Yeah because this is to speed up e.g. overlap computations of mode data at one frequency with the same mode data at a different frequency.

---

Ultimately, I will let you decide!

---

yeah, I think switching this one to if-based, but leaving the other one as try-catch seems like a good compromise at the moment. 

But here's some idea for the one falling back to xarray: if we know fairly broad, but maybe not exhaustive, conditions under which we expect with certainty the optimized calculation to complete successfully maybe it's worth to switching to still if-based branching. That is,
```
if (broad but not exhaustive ok):
    # direct calc
else:
    # xarray calc
```
Yes, we won't get the most performance in all possible cases, but hopefully in overwhelming majority. Again, not insisting on this though.

---

That's basically how I had it previously, I guess I just decided to be somewhat defensive against something randomly erroring in a way that I haven't anticipated and was not captured by tests. But I do agree with you guys that hiding errors is not great either.